src/rnp/fficli.cpp

```diff
@@ -1347,109 +1347,59 @@ cli_rnp_keys_matching_strings(cli_rnp_t *                     rnp,
  *
  *  @return true if path constructed successfully, or false otherwise
  **/
-static bool
-rnp_path_compose(
-  const char *dir, const char *subdir, const char *filename, char *res, size_t res_size)
+static std::string
+rnp_path_compose(const std::string &dir,
+                 const std::string &subdir,
+                 const std::string &filename)
 {
-    int pos;
-
-    /* checking input parameters for conrrectness */
-    if (!dir || !filename || !res) {
-        return false;
-    }
-
-    /* concatenating dir, subdir and filename */
-    if (strlen(dir) > res_size - 1) {
-        return false;
-    }
-
-    strcpy(res, dir);
-    pos = strlen(dir);
-
-    if (subdir) {
-        if ((pos > 0) && (res[pos - 1] != '/')) {
-            res[pos++] = '/';
+    std::string res = dir;
+    if (!subdir.empty()) {
+        if (!res.empty() && (res.back() != '/')) {
+            res.push_back('/');
         }
-
-        if (strlen(subdir) + pos > res_size - 1) {
-            return false;
-        }
-
-        strcpy(res + pos, subdir);
-        pos += strlen(subdir);
+        res.append(subdir);
     }
 
-    if ((pos > 0) && (res[pos - 1] != '/')) {
-        res[pos++] = '/';
-    }
-
-    if (strlen(filename) + pos > res_size - 1) {
-        return false;
-    }
-
-    strcpy(res + pos, filename);
-
-    return true;
-}
-
-/* helper function : get key storage subdir in case when user didn't specify homedir */
-static const char *
-rnp_cfg_get_ks_subdir(rnp_cfg_t *cfg, int defhomedir)
-{
-    const char *subdir;
-
-    if (!defhomedir) {
-        subdir = NULL;
-    } else {
-        if ((subdir = rnp_cfg_getstr(cfg, CFG_SUBDIRGPG)) == NULL) {
-            subdir = SUBDIRECTORY_RNP;
-        }
+    if (!res.empty() && (res.back() != '/')) {
+        res.push_back('/');
```

Comment:
... or even use some Windows function as `PathCombine` or `PathCchCombine` for Windows?

Reply:
@rrrooommmaaa I think that should be done in separate PR, which would check all of the path-related calls in the CLI.